libpod/image.go

```diff
@@ -50,29 +55,116 @@ type CopyOptions struct {
 // return will include the image, a false return will exclude it.
 type ImageFilter func(*storage.Image) bool
 
+type imagePullStruct struct {
+	imageName   string
+	tag         string
+	registry    string
+	hasRegistry bool
+	transport   string
+}
+
+func (ips imagePullStruct) returnFQName() string {
+	return fmt.Sprintf("%s%s/%s:%s", ips.transport, ips.registry, ips.imageName, ips.tag)
+
+}
+
+func getRegistriesToTry(image string) ([]types.ImageReference, error) {
+	var registries []types.ImageReference
+	var imageError = fmt.Sprintf("Unable to parse '%s'\n", image)
+	imgRef, err := reference.Parse(image)
+	if err != nil {
+		return registries, errors.Errorf(imageError)
```

Comment:
If we're returning an error, we usually don't bother returning any other variables - so `return nil, imageError` instead

Reply:
fixed